Add 3D Map View toolbars and animation (#5214)
qgis/QGIS-Documentation: docs/user_manual/introduction/qgis_gui.rst

===========

3D visualization support is offered through the 3D map view.
You create and open a 3D map view via :menuselection:`View -->`
|new3DMap| :menuselection:`New 3D Map View`.
A floating QGIS panel will appear. The panel can be docked.

To begin with, the 3D map view has the same extent and view as the
2D main map canvas. A set of navigation tools are available to turn
the view into 3D.

.. _figure_3dmapview:

.. figure:: img/3dmapview.png
   :align: center

   The 3D Map View dialog

> Following tools are provided at the top of the 3D map view panel:

Reviewer: ```suggestion
The following tools are provided at the top of the 3D map view panel:
```

* |pan| :sup:`Camera control`: moves the view, keeping the same angle
  and direction of the camera
* |zoomFullExtent| :sup:`Zoom Full`: resizes the view to the whole
  layers' extent
* |3dNavigation| :sup:`Toggle on-screen notification`: shows/hides the

>   navigation widget for a better control of the map view

Reviewer: ```suggestion
  navigation widget
```
As it was written, it does not say whether it is with or without this navigation widget that the control of the map view is better.

Author: Would
```suggestion
  navigation widget that can be used for a better control of the map view
```
(or replace "a better" with "an easy" or "a convenient") make it clearer? I think the navigation widget is meant to ease the navigation, compared to the previous "guess yourself how to proceed"

Reviewer: I  tried to modify.

* |identify| :sup:`Identify`: returns attributes of the clicked feature(s)
  -- More details at :ref:`identify`

> * |measure| :sup:`Measurement line`: measures horizontal distance between points

Reviewer: ```suggestion
* |measure| :sup:`Measurement line`: measures the horizontal distance between points
```

To reset the camera view, click the |zoomFullExtent| :sup:`Zoom Full`
button on the top of the 3D canvas panel.

.. _`create_animation`:

Creating an animation
---------------------

An animation is based on a set of keyframes - camera positions at particular times.
To create an animation:

#. Toggle on the |play| :sup:`Animations` tool, displaying the animation player
   widget
#. Click the |signPlus| :sup:`Add keyframe` button and enter a :guilabel:`Keyframe
   time` in seconds. The :guilabel:`Keyframe` combo box now displays the time set.

> #. Using the navigation tools, move the camera to a position to associate to

Reviewer: ```suggestion
#. Using the navigation tools, move the camera to the position to associate with
```

   the current keyframe time.

> #. Repeat previous steps to add as many keyframes time and position as necessary.

Reviewer: ```suggestion
#. Repeat the previous steps to add as many keyframes (with time and position) as necessary.
```
Or skip ` (with time and position)` altogether.

> #. Click |play| button to preview the animation. QGIS will generate a scene using

Reviewer: ```suggestion
#. Click the |play| button to preview the animation. QGIS will generate scenes using
```

>    the camera positions/rotations at set times and interpolating them in between

Reviewer: ```suggestion
   the camera positions/rotations at set times, and interpolating them in between
```

   these keyframes. Various :guilabel:`Interpolation` modes for animations are
   available (eg, linear, inQuad, outQuad, inCirc... -- more details at
   https://doc.qt.io/qt-5/qeasingcurve.html#EasingFunction-typedef).

   The animation can also be previewed by moving the time slider.
   Keeping the |draw| :sup:`Repeat` button pressed will repeatedly run the
   animation while clicking |play| stops a running animation.

> It's possible to browse the different views of the camera, using the

Reviewer: ```suggestion
It is possible to browse the different views of the camera, using the
```

:guilabel:`Keyframe` list. Whenever a time is active, changing the map view
will automatically update the associated position. You can also |symbologyEdit|

> :sup:`Edit keyframe` time or |signMinus| :sup:`Remove keyframe`.

Reviewer: ```suggestion
:sup:`Edit keyframe` or |signMinus| :sup:`Remove keyframe`.
```
Is `time` necessary here?

Author: This button allows to change the time only. Position is not concerned.

Reviewer: OK. I tried to rephrase.

> Click |fileSave| :sup:`Export animation frames` to generate a series of image

Reviewer: ```suggestion
Click |fileSave| :sup:`Export animation frames` to generate a series of images
```
